src/gui/generalsettings.cpp

```diff
@@ -456,7 +453,13 @@ void GeneralSettings::saveMiscSettings()
 
 void GeneralSettings::slotToggleLaunchOnStartup(bool enable)
 {
-    Theme *theme = Theme::instance();
+    ConfigFile configFile;
+    if (enable == Utility::hasLaunchOnStartup(Theme::instance()->appName())) {
+        return;
+    }
+
+    const auto theme = Theme::instance();
```

Comment:
why not move the declaration of `theme` before the `if` line ?
you would then be able to also use it in the `if`